src/blockdata/transaction.rs

```diff
@@ -702,17 +711,34 @@ impl Decodable for Transaction {
                             input,
                             output,
                             lock_time: Decodable::consensus_decode(d)?,
+                            mw_tx: None,
+                            is_hog_ex: false
                         })
                     }
                 }
                 8 | 9 => {
```

Comment:
The existing implementation isn't quite correct. `segwit_flag` should be treated as a bitmask, with each value indicating additional data to deserialize. So it should first check if the `1` bit is set, and if it is, deserialize input witnesses. Then it should check if the `8` bit is set, and if it is deserialize MWEB tx/hogex indicator.

If the value is `9` (0x00001001 == 0x00000001 | 0x00001000), that means there are both input witnesses (0x00000001) and MWEB tx data (0x00001000). In the current implementation, for a value of 9, we deserialize the MWEB tx/hogex indicator correctly (0x00001000) but we don't deserialize script witnesses (0x00000001). See: https://github.com/rust-litecoin/rust-litecoin/blob/5c7012f0a9953e95524a8a95cfd4dcebb0ee8770/src/blockdata/transaction.rs#L694-L696

Reply:
Made changes so that input witnesses are always deserialized when `1` bit is set